binary-search-tree/node-js/inventory-management.js

```diff
@@ -19,15 +23,40 @@ class BinarySearchTree {
     }
 
     search(id) {
-        console.error("you must implement search");
-    }
-
+            return this.searchRecursive(id, this.root);
+        }
+        // console.error("you must implement search");
+        // 1. Start from the root node
+        // 2. Compare the current node with the search value. Return early if found
+        // 3. Determine if we should go left or right
+        // 4. Update the current node based on the decision from (3). Rinse and repeat from (2)
+    searchRecursive(id, currentNode) {
+        if (currentNode === null) {
+            return null ;
+        }
+        else if(currentNode.val.id === id ) {
+            return currentNode.val;
+        }
+        else if (id > currentNode.val.id) {
+            return this.searchRecursive(id, currentNode.right);
+        }
+        else {
+            return this.searchRecursive(id, currentNode.left);
+        }
```

Comment:
Nice! I see that you're getting a better understanding of the exercise. This is good and it works, the `search` method test cases are passing. Here is some feedback to make it better:

There's a potential issue with conditional branches, specifically the `else` branch that could lead to unintended behavior in certain edge cases.

It's important to handle the cases you care about explicitly to avoid unexpected behavior. Currently, your code handles `id < currentNode.val.id` implicitly within the `else` branch. However, if `currentNode` is `undefined` (not `null`), this can lead to an undesired outcome. Specifically you'll get an an infinite loop (leaving it as an exercise for you to confirm).

To address this, I recommend adding an explicit `else if (id < currentNode.val.id)` branch

Reply:
ahh... ok. Originally i did have `else if (id < currentNode.val.id)` and then else { return;}